feat: implement unified JSON/text logging format

lngarrett opened the pull request:
This provides a unified logging system where users can switch between human-readable text logs and structured JSON logs across all output destinations: console, files, and WebSocket/UI debug view.

- Add logFormat setting to Gateway config (text/json)
- Update UI logger to respect global logFormat setting
- Add log format selector in Settings UI
- Update Debug view to handle both text and JSON log formats
- Implement driver logging configuration for JSON mode
- Use DailyRotateFile for JSON file transport to maintain symlink functionality
- Redirect driver internal file transport to /dev/null in JSON mode. See details below.
- Add WebSocket streaming for both text and JSON formats
- Add basic tests for logging format functionality


Why do I write a /dev/null file in JSON mode?

The zwave-js driver's internal logic creates a console transport when` (isTTY && !logToFile)` is true, but we needed` logToFile: true` to prevent console output in JSON mode. However, setting` logToFile: true` also creates an unwanted text file transport that writes to the same filename as our JSON logs, causing duplicate/conflicting file output.
Solution: I redirected the driver's internal file transport to `/dev/null` using `filename: /dev/null` in the driver's `logConfig`, which satisfies the driver's internal logic (preventing console transport creation) while discarding the unwanted text file output, allowing our custom JSON file transport to handle the actual file logging cleanly.

AlCalzone commented on api/lib/ZwaveClient.ts:
```diff
@@ -6946,6 +6942,97 @@ class ZwaveClient extends TypedEventEmitter<ZwaveClientEventCallbacks> {
+				...zwaveOptions.logConfig,
+				logToFile: true,       // This prevents internal console transport creation
+				forceConsole: false,   // This ensures no internal console transport
+				filename: '/dev/null', // Redirect driver's internal file transport to /dev/null
```
This will break on Windows systems, which we do support.

Also, instead of hacking with the different file/console settings, simply set `enabled: false`, which will disable the internal transports.
You can still provide your own which will not be affected by the `enabled` flag.

AlCalzone commented:
I double-checked. Formatting the messages happens outside of the transport format, in https://github.com/zwave-js/zwave-js/blob/266f47f49f804609d633313fe128a81c987732ff/packages/zwave-js/src/lib/log/Driver.ts#L113

We'd need to add an option `raw` to the log config in the driver options and evaluate that to skip the more complex formatting and instead `JSON.stringify` the [`logEntry`](https://github.com/zwave-js/zwave-js/blob/266f47f49f804609d633313fe128a81c987732ff/packages/zwave-js/src/lib/log/Driver.ts#L135) to get the message.

lngarrett commented on src/views/Debug.vue:
```diff
@@ -175,6 +175,48 @@ export default {
 			// no need to make this reative
 			this.prevScrollTop = scrollTop
 		},
+		formatLog(data) {
```
Right, I missed that sorry. I took Debug.vue back to original state and handled this via the backend. I left the original minimal formatting designed for text as it wasn't negatively impacting JSON.

lngarrett commented on api/lib/ZwaveClient.ts:
```diff
@@ -6946,6 +6942,120 @@ class ZwaveClient extends TypedEventEmitter<ZwaveClientEventCallbacks> {
+	 * Setup driver logging based on the configured format (text or JSON)
+	 * Uses the same logFormat setting as the gateway for consistency
+	 */
+	private setupDriverLogging(zwaveOptions: PartialZWaveOptions) {
```
I broke this up into more manageable functions and made it more DRY.

robertsLando reviewed (commented):
@AlCalzone Do you think this could be simplified ?

lngarrett commented:
@robertsLando @AlCalzone - Are there any other changes / feedback on this?

lngarrett committed:
feat: add JSON logging format support

- Add support for JSON logging format alongside existing text format
- JSON mode outputs structured JSON logs to console, files, and debug view
- Text mode continues to output human-readable text with ANSI colors
- Debug view displays raw format as received (JSON strings or formatted text)
- Logging format configurable via gateway settings (logFormat: 'json' | 'text')
- Both formats use consistent JSONTransport architecture
- Maintains backward compatibility with existing text logging behavior

This enables structured logging for better log parsing, monitoring,
and integration with log aggregation systems while preserving
the existing text format for human readability.

lngarrett commented:
@AlCalzone I think I've addressed everything with the exception of one open question:

- [x] Remvoed additional timestamp. If you agree the missing TZ info should come from driver then this should be good to go.
- [x] Message flattening is removed
- [ ] JSON/Text Toogle - I have one question above regarding what format to prefer in file, console, and UI formats when JSON logging is active.
